cpp/include/cudf/io/detail/json.hpp

```diff
@@ -33,7 +33,7 @@ namespace cudf::io::json::detail {
  *
  * @return cudf::table object that contains the array of cudf::column.
  */
-table_with_metadata read_json(std::vector<std::unique_ptr<cudf::io::datasource>>& sources,
+table_with_metadata read_json(host_span<std::unique_ptr<datasource>> sources,
```

Comment:
Out of scope for this PR, but a `host_span<unique_ptr>` feels off when it comes to ownership semantics. This pattern seems prevalent in cuIO though. Also I don't know if there's a nice idiomatic way to write this otherwise though since most implicit conversions won't quite work and there's limited interop between implicit conversions and template deduction.

Reply:
Yeah, really sends a mixed message :D
Surely we can lose the `unique_ptr`s somewhere along the call chain...